packages/smooth_app/lib/pages/folksonomy/folksonomy_card.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/helpers/product_cards_helper.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_page.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_provider.dart';
+import 'package:smooth_app/pages/folksonomy/tag.dart';
+import 'package:smooth_app/pages/product/common/product_refresher.dart';
+import 'package:smooth_app/themes/constant_icons.dart';
+
+class FolksonomyCard extends StatelessWidget {
+  const FolksonomyCard(this.product);
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider<FolksonomyProvider>(
+      create: (_) => FolksonomyProvider(product.barcode!),
+      child: Card(product),
+    );
+  }
+}
+
+class Card extends StatelessWidget {
+  const Card(this.product);
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    return InkWell(
+      onTap: () async {
+        if (!await ProductRefresher().checkIfLoggedIn(
+          context,
+          isLoggedInMandatory: true,
+        )) {
+          return;
+        }
+        if (!context.mounted) {
+          return;
+        }
+
+        Navigator.of(context)
+            .push(
+              MaterialPageRoute<void>(
+                builder: (BuildContext context) => FolksonomyPage(product),
+              ),
+            )
+            .then((void completion) => <Set<Future<void>>>{
+                  if (context.mounted)
```

Comment:
It's totally OK to use the callback for the `Future` here.
But generally speaking, for the project, we prefer the `async/await` paradigm.